dance/datasets/spatial.py

```diff
@@ -175,8 +234,10 @@ def _load_raw_data(self) -> Tuple[pd.DataFrame, ...]:
                 continue
             elif ext == ".csv":
                 raw_data_dict[filename] = pd.read_csv(filepath, header=0, index_col=0)
+                raw_data_dict[filename].index = raw_data_dict[filename].index.astype(str)
             elif ext == ".h5ad":
                 raw_data_dict[filename] = sc.read_h5ad(filepath).to_df()
+                raw_data_dict[filename].index = raw_data_dict[filename].index.astype(str)
```

Comment:
You can also potentially move this specific index type conversion (and the one above) out of the if clause to reduce reduandant code.

Reply:
I have fixed it